src/clients/ApplicationClient.js

```diff
@@ -704,7 +704,130 @@ export default class ApplicationClient extends BaseClient {
     this.log.debug("[ApplicationClient] getAllDevices() - BULK");
     return this.callApi('GET', 200, true, ["bulk", "devices"], null, params);
   }
-
+	
+   /**
+   * Gateway Access Control (Beta)
+   * The methods in this section follow the documentation listed under the link:
+   * https://console.ng.bluemix.net/docs/services/IoT/gateways/gateway-access-control.html#gateway-access-control-beta-
+   * Involves the following sections from the above mentioned link:
+   * Assigning a role to a gateway
+   * Adding devices to and removing devices from a resource group
+   * Finding a resource group
+   * Querying a resource group
+   * Creating and deleting a resource group
+   * Updating group properties
+   * Retrieving and updating device properties
+   * 
+   */
+  
+	//getGatewayGroup(gatewayId)
+	//updateDeviceRoles(deviceId, roles[])
+	//getAllDevicesInGropu(groupId)
+	//addDevicesToGroup(groupId, deviceList[])
+	//removeDevicesFromGroup(groupId, deviceList[])
+  
+  getGatewayGroup(gatewayId){
+    this.log.debug("[ApplicationClient] getGatewayGroup("+gatewayId+")");
+    return this.callApi('GET', 200, true, ['authorization', 'devices' , gatewayId], null);
+  }
+  
+  updateDeviceRoles(deviceId, roles){
```

Comment:
I would recommend duplicating this function so that it is clear that it applies to both devices and gateways:

`updateDeviceRoles(deviceId, roles)`
`updateGatewayRoles(gatewayId, roles)`

updateGatewayRoles can simply call updateDeviceRoles